base/shell/cmd/batch.c

```diff
@@ -408,17 +409,9 @@ INT Batch(LPTSTR fullname, LPTSTR firstword, LPTSTR param, PARSED_COMMAND *Cmd)
     /* Perform top-level batch initialization */
     if (bTopLevel)
     {
-        TCHAR *dot;
-
-        /* Default the top-level batch context type to .BAT */
-        BatType = BAT_TYPE;
-
-        /* If this is a .CMD file, adjust the type */
-        dot = _tcsrchr(bc->BatchFilePath, _T('.'));
-        if (dot && (!_tcsicmp(dot, _T(".cmd"))))
-        {
-            BatType = CMD_TYPE;
-        }
+        /* Is the top-level batch context type .CMD or .BAT? */
+        TCHAR *dotext = _tcsrchr(bc->BatchFilePath, _T('.'));
```

Comment:
```suggestion
        /* Default the top-level batch context type
         * to .BAT, unless this is a .CMD file */
        PTCHAR dotext = _tcsrchr(bc->BatchFilePath, _T('.'));
```
Besides the NT-ification of the pointer to TCHAR made here,
I suggest the comment change to emphasize more (as was the intent in the original code) the fact that the batch type _always default_ to BAT; that is, unless the file is .CMD.